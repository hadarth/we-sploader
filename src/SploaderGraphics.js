import React from 'react';
import PropTypes from 'prop-types'

const SploaderGraphics = ({percent, SploaderSvg}) => {
  const strokeDashoffset = 100 - percent
  return (
    <svg ref={SploaderSvg} className='spinit' xmlns='http://www.w3.org/2000/svg' viewBox='0 0 200 200'>
      <circle className='base' cx='100' cy='100' r='90' />
      <circle style={{ strokeDashoffset }} className='blue' cx='100' cy='100' r='90' pathLength="100"/>
    </svg>
  )
}

SploaderGraphics.propTypes = {
  percent: PropTypes.number,
  SploaderSvg: PropTypes.shape({
    current: PropTypes.object
  })
}


export default SploaderGraphics;
