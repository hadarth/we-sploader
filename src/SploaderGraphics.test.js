import React from 'react';
import SploaderGraphics from './SploaderGraphics';
import Enzyme, { shallow } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

import enzymeSerializer from 'enzyme-to-json/serializer';
expect.addSnapshotSerializer(enzymeSerializer);

const mockRef = {current: {}}
const pecent = 10

describe("SploaderGraphics", () => {
  test('Should render and calculate percentage', () => {
    const wrapper = shallow(<SploaderGraphics percent={pecent} SploaderSvg={mockRef} />)
    expect(wrapper).toMatchSnapshot()
  })
});
