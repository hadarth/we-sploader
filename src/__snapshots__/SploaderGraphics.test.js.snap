// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`SploaderGraphics Should render and calculate percentage 1`] = `
<svg
  className="spinit"
  viewBox="0 0 200 200"
  xmlns="http://www.w3.org/2000/svg"
>
  <circle
    className="base"
    cx="100"
    cy="100"
    r="90"
  />
  <circle
    className="blue"
    cx="100"
    cy="100"
    pathLength="100"
    r="90"
    style={
      Object {
        "strokeDashoffset": 90,
      }
    }
  />
</svg>
`;
